tests/react/vanilla-utils/atomWithStorage.test.tsx

```diff
@@ -353,20 +353,49 @@ describe('atomWithStorage (in non-browser environment)', () => {
   }
 
   const addEventListener = window.addEventListener
+  const localStorage = window.localStorage
+  const sessionStorage = window.sessionStorage
 
   beforeAll(() => {
     ;(window as any).addEventListener = undefined
+    Object.defineProperty(window, 'localStorage', {
+      get() {
+        throw new Error('localStorage is not available.')
+      },
+    })
+    Object.defineProperty(window, 'sessionStorage', {
+      get() {
+        throw new Error('localStorage is not available.')
+      },
+    })
   })
 
   afterAll(() => {
     window.addEventListener = addEventListener
+    Object.defineProperty(window, 'localStorage', {
+      get() {
+        return localStorage
+      },
+    })
+    Object.defineProperty(window, 'sessionStorage', {
+      get() {
+        return sessionStorage
+      },
+    })
   })
 
   it('createJSONStorage with undefined window.addEventListener', async () => {
     const storage = createJSONStorage(() => asyncDummyStorage)
-
     expect(storage.subscribe).toBeUndefined()
   })
+
+  it('createJSONStorage with localStorage', async () => {
+    expect(() => createJSONStorage(() => window.localStorage)).not.toThrow()
```

Comment:
What we should check is not only createJSONStorage, but also atomWithStorage.
You want to confirm if the test works as expected by removing the try-catch we added.

Reply:
I tested with the try/catch removed and it did throw an error:

![CleanShot 2024-05-30 at 15 17 44@2x](https://github.com/pmndrs/jotai/assets/29674767/68cd797f-308b-43ec-8b4b-8c7e6e9985f7)

should we add a new test or modify this one in you opinion?

Reply:
thanks. that's my oversight. yeah, the current test seems good. no need to add new one.

Reply:
oh, I think what we need to test, if possible, is `createJSONStorage()`. Do you have any idea?

Reply:
you mean instead of `createJSONStorage(() => window.localStorage)` we should test `createJSONStorage()`?

Reply:
Yes, if possible, we want to test both. Otherwise, it's fine for now.

Reply:
I simply added another expect to check for it, but I have a feeling you mean something else.
there is also a `console.warn` happening here due to the checking we have in `createJSONStorage`. is it ok?

here is a screenshot:

![CleanShot 2024-06-02 at 13 06 43@2x](https://github.com/pmndrs/jotai/assets/29674767/26ffa56a-6b20-4416-93dd-57a2582b334a)

Reply:
Thanks.

`except(() => createJSONStorage()).not.toThrow()` seems good. (It fails if we remove the fix, right?)

As for the console.warn, let's mock it for this test and hide the warning.

Reply:
> It fails if we remove the fix, right?

Yes


Also, I added `console.warn` patching in the last commit, please check it.